Frameworks/Foundation/NSProxy.mm

```diff
@@ -14,220 +14,317 @@
  @Notes
 */
 - (void)dealloc {
-    UNIMPLEMENTED();
+    object_dispose(self);
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)finalize {
-    UNIMPLEMENTED();
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)forwardInvocation:(NSInvocation*)anInvocation {
-    UNIMPLEMENTED();
+    [NSException raise:NSInvalidArgumentException
+                format:@"[NSProxy forwardInvocation:] was called directly. It must be overridden in a subclass."];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (NSMethodSignature*)methodSignatureForSelector:(SEL)aSelector {
-    UNIMPLEMENTED();
-    return StubReturn();
+    [NSException raise:NSInvalidArgumentException
+                format:@"[NSProxy methodSignatureForSelector:] was called directly. It must be overridden in a subclass."];
+    return nil;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 + (BOOL)respondsToSelector:(SEL)aSelector {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return class_respondsToSelector(self, aSelector);
+}
+
+// clang format doesn't play well with 'class' here, so turn it off
+// clang-format off
+
+/**
+ @Status Interoperable
+ @Notes
+*/
++ (Class)class {
+    return self;
+}
+
+
+/**
+ @Status Interoperable
+*/
++ (Class)superclass {
+    return class_getSuperclass(self);
+}
+
+// clang-format on
+
+/**
+ @Status Interoperable
+*/
+
+- (NSString*)description {
+    return [NSString stringWithFormat:@"<%@: %p>", NSStringFromClass([self class]), self];
+}
+
+/**
+ @Status Interoperable
+*/
+- (NSString*)debugDescription {
+    return [self description];
 }
 
+#pragma endregion
+
+#pragma region NSObject protocol functions
+
+// clang format doesn't play well with 'class' here, so turn it off
+// clang-format off
+
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
-+ (Class) class {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (Class)class {
+    return object_getClass(self);
 }
 
-    /**
-     @Status Stub
-     @Notes
-    */
-    - (Class) class {
-    UNIMPLEMENTED();
-    return StubReturn();
+/**
+ @Status Interoperable
+ @Notes
+*/
+- (Class)superclass {
+    return class_getSuperclass(object_getClass(self));
 }
 
-    /**
-     @Status Stub
-     @Notes
-    */
-    - (BOOL)isEqual : (id)anObject {
-    UNIMPLEMENTED();
-    return StubReturn();
+// clang-format on
+
+/**
+ @Status Interoperable
+ @Notes
+*/
+- (BOOL)isEqual:(id)anObject {
+    return self == anObject;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
+*/
+- (NSUInteger)hash {
+    return reinterpret_cast<NSUInteger>(self);
+}
+
+/**
+ @Status Interoperable
  @Notes
 */
 - (instancetype)self {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return (id)self;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (BOOL)isKindOfClass:(Class)aClass {
-    UNIMPLEMENTED();
-    return StubReturn();
+    Class forwardClass = static_cast<Class>([self performSelector:@selector(class)]);
```

Comment:
it's the contract unfortunately :( [NSProxy class] does not return a forwarded class